cloudsmith_cli/cli/commands/metrics.py

```diff
@@ -0,0 +1,158 @@
+
+
+def _print_total_usage_table(opts, data):
+    """ Print total usage metrics as a table. """
+    headers = ["Total Tokens", "Active Tokens", "Inactive Tokens", "Bandwidth Used"]
+
+    click.echo(click.style("\nUsage Totals", bold=True, fg="white"))
+    click.echo(
+        "---------------------------------------------------------------", nl=False
+    )
+
+    rows = []
+    rows.append(
+        [
+            click.style(str(data.totals.get("tokens", 0)), fg="blue"),
+            click.style(str(data.totals.get("active_tokens", 0)), fg="green"),
+            click.style(str(data.totals.get("inactive_tokens", 0)), fg="red"),
+            click.style(str(data.totals.get("bandwidth_used", 0)), fg="white"),
+        ]
+    )
+
+    if data:
+        click.echo()
+        utils.pretty_print_table(headers, rows)
+
+    click.echo()
+
+
+def _print_bandwith_usage_table(opts, data):
+    """ Print bandwidth usage metrics as a table. """
+    headers = ["Lowest usage", "Average usage", "Highest usage"]
+
+    click.echo(click.style("Bandwidth Usage Statistics", bold=True, fg="white"))
+    click.echo(
+        "---------------------------------------------------------------", nl=False
+    )
+
+    rows = []
+    rows.append(
+        [
+            click.style(str(data.bandwidth_per_token.get("lowest", 0)), fg="white"),
+            click.style(str(data.bandwidth_per_token.get("average", 0)), fg="white"),
+            click.style(str(data.bandwidth_per_token.get("highest", 0)), fg="white"),
+        ]
+    )
+
+    if data:
+        click.echo()
+        utils.pretty_print_table(headers, rows)
+
+    click.echo()
+
+
+def print_metrics(opts, data):
+    """ Print metrics as a table or output in another format. """
+    _print_total_usage_table(opts, data)
+    _print_bandwith_usage_table(opts, data)
+
+
+@main.group(cls=command.AliasGroup, name="metrics", aliases=["metrics"])
+@decorators.common_cli_config_options
+@decorators.common_cli_output_options
+@decorators.common_api_auth_options
+@decorators.initialise_api
+@click.pass_context
+def metrics(ctx, opts):  # pylink: disable=unused-argument
+    """
+    Retrieve Metrics.
+
+    See the help for subcommands for more information on each.
+    """
+
+
+@metrics.command(name="usage", aliases=["usage"])
+@decorators.common_cli_config_options
+@decorators.common_cli_output_options
+@decorators.common_api_auth_options
+@decorators.initialise_api
+@click.argument(
+    "owner_repo",
+    metavar="OWNER/REPO",
+    callback=validators.validate_owner_repo,
+    required=True,
+)
+@click.option(
+    "--tokens",
+    type=str,
+    required=False,
+    help="A comma seperated list of entitlement tokens (identifiers). Each identifier is exactly 12 characters in "
+    "length and only contain alphanumerics. If a list is not specified then "
+    "all entitlement tokens will be included for a given repository.",
+    callback=validators.validate_optional_tokens,
+)
+@click.option(
+    "--start",
```

Comment:
NAB: Just wondering if epoch is the easiest way for users to use this. Might be nicer to consume a UTC string or something, that we convert to epoch in the CLI before forwarding? Just a thought - would like to see what others think.

Reply:
Yeah, same. We have some nice examples in the server-side code of parsing based on a human-specific datetime, and it would be nice to replicate here. Advice for user-facing stuff: "Don't Make Me Think." (and I can't think of an epoch timestamp without using a tool to look it up). ;-)

Reply:
will update 👍

Reply:
`
cloudsmith metrics usage cloudsmith/testing-private --start=2020-01-01T00:00:00Z --finish=2020-12-01T00:00:00Z
`

<img width="903" alt="Screenshot 2020-09-02 at 10 35 36" src="https://user-images.githubusercontent.com/1443700/91965200-140ae100-ed08-11ea-9b7f-6bbb1263ade1.png">